core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -503,24 +537,29 @@ object StorageTool extends Logging {
   }
 
   def formatCommand(
-    stream: PrintStream,
-    directories: Seq[String],
-    metaProperties: MetaProperties,
-    metadataVersion: MetadataVersion,
-    ignoreFormatted: Boolean
-  ): Int = {
+                     stream: PrintStream,
+                     directories: Seq[String],
+                     metaProperties: MetaProperties,
+                     metadataVersion: MetadataVersion,
+                     ignoreFormatted: Boolean,
+                     advertisedListenerEndpoints: scala.collection.Seq[kafka.cluster.EndPoint],
+                     controllersQuorumVoters: String
+                   ): Int = {
     val bootstrapMetadata = buildBootstrapMetadata(metadataVersion, None, "format command")
-    formatCommand(stream, directories, metaProperties, bootstrapMetadata, metadataVersion, ignoreFormatted)
+    formatCommand(stream, directories, metaProperties, bootstrapMetadata, metadataVersion, ignoreFormatted,
+      advertisedListenerEndpoints, controllersQuorumVoters)
   }
 
   def formatCommand(
-    stream: PrintStream,
-    directories: Seq[String],
-    metaProperties: MetaProperties,
-    bootstrapMetadata: BootstrapMetadata,
-    metadataVersion: MetadataVersion,
-    ignoreFormatted: Boolean
-  ): Int = {
+                     stream: PrintStream,
+                     directories: Seq[String],
+                     metaProperties: MetaProperties,
+                     bootstrapMetadata: BootstrapMetadata,
+                     metadataVersion: MetadataVersion,
+                     ignoreFormatted: Boolean,
+                     advertisedListenerEndpoints: scala.collection.Seq[kafka.cluster.EndPoint],
+                     controllersQuorumVoters: String
```

Comment:
Did you consider making the input to this method just `voters: VoterSet` instead of `advertisedListenerEndpoints` and `controllerQuorrumVoters`.

When the --standalone flag is used the set of voters can be constructed from `VoterSet.fromMap(createStandaloneVoterMap(nodeId, nodeDirectoryId, config.effectiveAdvertisedControllerListeners))`.

When the --controller-quorum-voters flag is used the set of voters can be constructed from `VoterSet.fromMap(parseVotersToMap(controllersQuorumVoters))`